specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimdeployment.json

```diff
@@ -0,0 +1,1400 @@
+        "redirectUri": {
+          "type": "string",
+          "description": "Redirect URL to the Publisher Portal containing the SSO token."
+        }
+      },
+      "description": "The response of the GetSsoToken operation."
+    },
+    "ApiManagementServiceCheckNameAvailabilityParameters": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name to check for availability."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "Parameters supplied to the CheckNameAvailability operation."
+    },
+    "ApiManagementServiceNameAvailabilityResult": {
+      "properties": {
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "True if the name is available and can be used to create a new API Management service; otherwise false.",
+          "readOnly": true
+        },
+        "message": {
+          "type": "string",
+          "description": "If reason == invalid, provide the user with the reason why the given name is invalid, and provide the resource naming requirements so that the user can select a valid name. If reason == AlreadyExists, explain that <resourceName> is already in use, and direct them to select a different name.",
+          "readOnly": true
+        },
+        "reason": {
+          "type": "string",
+          "description": "Invalid indicates the name provided does not match the resource provider’s naming requirements (incorrect length, unsupported characters, etc.)  AlreadyExists indicates that the name is already in use and is therefore unavailable.",
+          "enum": [
+            "Valid",
+            "Invalid",
+            "AlreadyExists"
+          ],
+          "x-ms-enum": {
+            "name": "NameAvailabilityReason",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Response of the CheckNameAvailability operation."
+    },
+    "ApiManagementServiceApplyNetworkConfigurationParameters": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Location of the Api Management service to update for a multi-region service. For a service deployed in a single region, this parameter is not required."
+        }
+      },
+      "description": "Parameter supplied to the Apply Network configuration operation."
+    },
+    "ApiManagementServiceUploadCertificateParameters": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Hostname type.",
+          "enum": [
+            "Proxy",
+            "Portal",
+            "Management",
+            "Scm"
+          ],
+          "x-ms-enum": {
+            "name": "HostnameType",
+            "modelAsString": false
+          }
+        },
+        "certificate": {
+          "type": "string",
+          "description": "Base64 Encoded certificate."
+        },
+        "certificate_password": {
```

Comment:
>certificate_password [](start = 9, length = 20)

It is recommended to use camelCase

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/1881

Reply:
OKok :)